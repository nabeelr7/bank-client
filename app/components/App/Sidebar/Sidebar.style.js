/**
 *
 * SidebarWrapper
 *
 */

import styled from 'styled-components';
import { Layout } from 'antd';
// import { PRIMARY_RED, SECONDARY_RED } from 'utils/colors';
import { media } from 'utils';

export const StyledSidebar = styled(Layout.Sider)`
  overflow: auto;
  height: 100vh;
  position: fixed;
  left: 0;
  transition: all 0.1s ease-in-out;
  box-shadow: 4px 0px 8px -3px rgba(17, 17, 17, 0.6);
  display: none;
  overflow-x: hidden;

  ${media.tablet`
      display: block;
    `}
`;
