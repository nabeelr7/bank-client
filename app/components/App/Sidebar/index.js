/**
 *
 * Sidebar
 *
 */

import React from 'react';

import { useSelector } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { makeSelectIsCollapsedSidebar } from 'containers/App/selectors';
import Navigation from 'components/App/Navigation';
import Footer from 'components/App/Footer';
import { StyledSidebar } from './Sidebar.style';
import Logo from '../Logo';

// import Navigation from 'components/App/Navigation';
// import Footer from 'components/App/Footer';
// import SidebarWrapper from './SidebarWrapper';
// import LogoWrapper from './LogoWrapper';

const stateSelector = createStructuredSelector({
  isCollapsedSidebar: makeSelectIsCollapsedSidebar(),
});

export default function Sidebar() {
  const { isCollapsedSidebar } = useSelector(stateSelector);

  return (
    <StyledSidebar
      trigger={null}
      collapsible
      collapsed={isCollapsedSidebar}
      width={250}
    >
      <Logo />
      <Navigation />
      <Footer />
    </StyledSidebar>
  );
}
