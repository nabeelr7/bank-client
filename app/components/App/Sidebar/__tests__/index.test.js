import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import configureStore from 'configureStore';
import Sidebar from '../index';

describe('<Sidebar />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Sidebar />
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
