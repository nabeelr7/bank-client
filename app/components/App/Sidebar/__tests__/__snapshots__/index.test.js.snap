// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should match the snapshot 1`] = `
<aside
  class="Sidebarstyle__StyledSidebar-vp7xt9-0 kjWaBn ant-layout-sider ant-layout-sider-light"
  style="flex: 0 0 250px; max-width: 250px; min-width: 250px; width: 250px;"
>
  <div
    class="ant-layout-sider-children"
  >
    <div
      class="Logostyle__StyledLogo-sgm567-0 mCXGT"
    >
      <img
        alt="Bank Application"
        height="auto"
        src="IMAGE_MOCK"
        width="150"
      />
    </div>
    <ul
      class="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
      role="menu"
    >
      <li
        class="ant-menu-item Navigationstyle__StyledMenuItem-sc-12jopoh-0 dVGSQr ant-menu-item-selected"
        role="menuitem"
        style="padding-left: 24px;"
      >
        <span
          aria-label="desktop"
          class="anticon anticon-desktop Navigationstyle__StyledDesktopOutlined-sc-12jopoh-1 kvcTwU"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="desktop"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M928 140H96c-17.7 0-32 14.3-32 32v496c0 17.7 14.3 32 32 32h380v112H304c-8.8 0-16 7.2-16 16v48c0 4.4 3.6 8 8 8h432c4.4 0 8-3.6 8-8v-48c0-8.8-7.2-16-16-16H548V700h380c17.7 0 32-14.3 32-32V172c0-17.7-14.3-32-32-32zm-40 488H136V212h752v416z"
            />
          </svg>
        </span>
        <span>
          Dashboard
        </span>
      </li>
      <li
        class="ant-menu-item Navigationstyle__StyledMenuItem-sc-12jopoh-0 dVGSQr"
        role="menuitem"
        style="padding-left: 24px;"
      >
        <span
          aria-label="wallet"
          class="anticon anticon-wallet Navigationstyle__StyledWalletOutlined-sc-12jopoh-2 vlSWa"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="wallet"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 464H528V448h312v128zm0 264H184V184h656v200H496c-17.7 0-32 14.3-32 32v192c0 17.7 14.3 32 32 32h344v200zM580 512a40 40 0 1080 0 40 40 0 10-80 0z"
            />
          </svg>
        </span>
        <span>
          Payment
        </span>
      </li>
      <li
        class="ant-menu-item Navigationstyle__StyledMenuItem-sc-12jopoh-0 dVGSQr"
        role="menuitem"
        style="padding-left: 24px;"
      >
        <span
          aria-label="history"
          class="anticon anticon-history Navigationstyle__StyledHistoryOutlined-sc-12jopoh-3 hjiJcm"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="history"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M536.1 273H488c-4.4 0-8 3.6-8 8v275.3c0 2.6 1.2 5 3.3 6.5l165.3 120.7c3.6 2.6 8.6 1.9 11.2-1.7l28.6-39c2.7-3.7 1.9-8.7-1.7-11.2L544.1 528.5V281c0-4.4-3.6-8-8-8zm219.8 75.2l156.8 38.3c5 1.2 9.9-2.6 9.9-7.7l.8-161.5c0-6.7-7.7-10.5-12.9-6.3L752.9 334.1a8 8 0 003 14.1zm167.7 301.1l-56.7-19.5a8 8 0 00-10.1 4.8c-1.9 5.1-3.9 10.1-6 15.1-17.8 42.1-43.3 80-75.9 112.5a353 353 0 01-112.5 75.9 352.18 352.18 0 01-137.7 27.8c-47.8 0-94.1-9.3-137.7-27.8a353 353 0 01-112.5-75.9c-32.5-32.5-58-70.4-75.9-112.5A353.44 353.44 0 01171 512c0-47.8 9.3-94.2 27.8-137.8 17.8-42.1 43.3-80 75.9-112.5a353 353 0 01112.5-75.9C430.6 167.3 477 158 524.8 158s94.1 9.3 137.7 27.8A353 353 0 01775 261.7c10.2 10.3 19.8 21 28.6 32.3l59.8-46.8C784.7 146.6 662.2 81.9 524.6 82 285 82.1 92.6 276.7 95 516.4 97.4 751.9 288.9 942 524.8 942c185.5 0 343.5-117.6 403.7-282.3 1.5-4.2-.7-8.9-4.9-10.4z"
            />
          </svg>
        </span>
        <span>
          History
        </span>
      </li>
      <li
        class="ant-menu-item Navigationstyle__StyledMenuItem-sc-12jopoh-0 dVGSQr"
        role="menuitem"
        style="padding-left: 24px;"
      >
        <span
          aria-label="setting"
          class="anticon anticon-setting Navigationstyle__StyledSettingOutlined-sc-12jopoh-4 cjaXZl"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="setting"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 009.3-35.2l-.9-2.6a443.74 443.74 0 00-79.7-137.9l-1.8-2.1a32.12 32.12 0 00-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 00-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 00-25.8 25.7l-15.8 85.4a351.86 351.86 0 00-99 57.4l-81.9-29.1a32 32 0 00-35.1 9.5l-1.8 2.1a446.02 446.02 0 00-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 00-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0035.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0025.8 25.7l2.7.5a449.4 449.4 0 00159 0l2.7-.5a32.05 32.05 0 0025.8-25.7l15.7-85a350 350 0 0099.7-57.6l81.3 28.9a32 32 0 0035.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 01-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 01-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 01624 502c0 29.9-11.7 58-32.8 79.2z"
            />
          </svg>
        </span>
        <span>
          Settings
        </span>
      </li>
    </ul>
    <footer
      class="Footerstyle__StyledFooter-a5b9l3-0 gMHClc"
    >
      <strong>
        Bank Application v2.0
      </strong>
       | 22.05.20
    </footer>
  </div>
</aside>
`;
